File: src/LogWatcher/Domain/Models/SearchSet.cs
using System;
using System.Collections.Generic;

namespace Domain.Models
{
    public class SearchSet
    {
        public List<SourceSystem> SourceSystems { get; set; } = new List<SourceSystem>();

        public string KeyWordList { get; set; }

        public (DateTime Start, DateTime End) SearchPeriod { get; set; }
    }
}

File: src/LogParser/searchset.py
from pydantic import BaseModel
from typing import List

class SourceSystem(BaseModel):
    ID: int
    Name: str
    SourceFolder: str
    LineTemplate: str | None = None

class SearchPeriod(BaseModel):
    Item1: str
    Item2: str

class SearchSet(BaseModel):
  SourceSystems: List[SourceSystem]
  KeyWordList: str | None = None
  SearchPeriod: SearchPeriod

